galley/cmd/galley/cmd/root.go

```diff
@@ -18,27 +18,42 @@ import (
 	"flag"
 	"fmt"
 	"time"
-
 	"github.com/spf13/cobra"
-	"github.com/spf13/cobra/doc"
-
 	"istio.io/istio/galley/cmd/shared"
-	"istio.io/istio/pkg/collateral"
 	"istio.io/istio/pkg/log"
+	"istio.io/istio/galley/pkg/server"
+	galleyProbe "istio.io/istio/galley/pkg/probe"
+	"istio.io/istio/galley/pkg/crd/validation"
+	istiocmd "istio.io/istio/pkg/cmd"
+	"istio.io/istio/pkg/collateral"
+	"github.com/spf13/cobra/doc"
 	"istio.io/istio/pkg/version"
+
+	"istio.io/istio/pkg/probe"
 )
 
 var (
 	flags = struct {
 		kubeConfig   string
 		resyncPeriod time.Duration
+		noValidation         bool
+		noServer             bool
+		livenessProbeOptions *probe.Options
+		readinessProbeOptions *probe.Options
 	}{}
 
 	loggingOptions = log.DefaultOptions()
 )
 
+
+
+
 // GetRootCmd returns the root of the cobra command-tree.
-func GetRootCmd(args []string, printf, fatalf shared.FormatFn) *cobra.Command {
+func GetRootCmd(args []string, printf, fatalf shared.FormatFn,) *cobra.Command {
+
+	var sa = server.DefaultArgs()
+	var vc = validation.DefaultWebhooktParams()
+
```

Comment:
We could simplify this a bit by hooking `PersistantFlags.StringVar()` directly to the server and validation args. e.g.

```golang
func GetRootCmd(...) {
    var (
        serverArgs = server.DefaultArgs()
        validationARgs = validadtion.DefaultArgs()
    )

    rootCmd := &cobra.Command{ ... }

    rootCmd.PersistentFlags().BoolVar(&serverArgs.server, "server", "Run galley server mode", server.Args.server)
    // etc
}
```

This would also keep the `server.DefaultArgs`() and default flag values in sync.

Reply:
Done~